backend/src/main/java/com/neuefische/team2/backend/restaurant/RestaurantService.java

```diff
@@ -61,4 +63,29 @@ public void deleteRestaurant(String id) {
         this.findRestaurantById(id);
         restaurantRepository.deleteById(id);
     }
+
+    public Restaurant addCommentToRestaurant(String id, String commentText) throws ResourceNotFoundException {
+        logger.info("Trying to add comment to restaurant with ID {}", id);
+
+        Restaurant restaurant = this.findRestaurantById(id);
+        List<Restaurant.Comment> updatedComments = new ArrayList<>(restaurant.comments() != null ? restaurant.comments() : new ArrayList<>());
```

Comment:
It it not very nice to have such a logic inside a type definition. Is there a better way to do this? Why do you need this logic anyway and why an ArrayList inside an ArrayList?